submissions/Yasya23/tiny-js-oop/index.js

```diff
@@ -1,38 +1,35 @@
 class Inhabitants {
-  constructor(species, name, gender) {
+  constructor(species, name, gender, saying) {
     this.species = species;
     this.name = name;
     this.gender = gender;
+    this.saying = saying;
   }
 
   showProperties() {
-    return `${this.species}; ${this.name}; ${this.gender}`;
+    return [this.species, this.name, this.gender];
   }
 }
 
 class Human extends Inhabitants {
   constructor(name, gender, saying) {
-    super("human", name, gender);
+    super("human", name, gender, saying);
     this.hands = 2;
     this.legs = 2;
-    this.saying = saying;
   }
 
   showProperties() {
-    return `${super.showProperties()}; ${this.hands}; ${this.legs}; ${
-      this.saying
-    }`;
+    return [...super.showProperties(), this.hands, this.legs, this.saying];
```

Comment:
Here and line 32 below:
- repetition
- also a shared property should be handled by the class that owns it

Reply:
Fixed it but I`m not sure about my decision. If it's incorrect please give me advice.

Reply:
That's a good start.
Your previous method version returned a string. Now it returns array not joined into string. Why?

Reply:
I'm a little confused. The join method returns a string from the array of elements. If I add it here then it will return a string.

`  showProperties() {
    return [
      ...super.showProperties().slice(0, 3),
      this.paws,
      ...super.showProperties().slice(3, 4),
    ].join("; ");
  }`

But I don’t understand the difference, because at the end I still iterate over the array and add a join. 

`inhabitants.forEach(inhabitant => print(inhabitant.showProperties().join("; ")));`

Apparently, there is something I don't understand.

Reply:
My question was about reasons behind changing from string to array. It is a significant change.

Reply:
Both approaches may be valid. Want to understand reasoning.

Reply:
Because of that post)

This and similar fragments below:
Benefit from Array methods. This will make code DRYer.


For myself, I realized that I need to create an array instead of a string. But now I understand that I could iterate through the array of values ​​​​using the map method and add a join, and not write a semicolon after each value. Something like this.

`["species", "name", "gender", "saying"]
  .map((property) => this.property)
  .join("; ");
`

Only working with an array for me here looks easier if I want to insert a value in its middle.